languages/elixir/exercises/concept/bitstrings/.meta/example.ex

```diff
@@ -0,0 +1,51 @@
+defmodule DNA do
+  @moduledoc """
+  Example solution for the `bitstrings` exercise.
+
+  Written by Tim Austin, tim@neenjaw.com, June 2020.
+
+   | NucleicAcid | Bits |
+   | ----------- | ---- |
+   |     A       | 0001 |
+   |     C       | 0010 |
+   |     G       | 0100 |
+   |     T       | 1000 |
+   |    " "      | 0000 |
+  """
+
+  @pairs [
+    {?\s, 0b0000},
+    {?A, 0b0001},
+    {?C, 0b0010},
+    {?G, 0b0100},
+    {?T, 0b1000},
+  ]
+
+  @pairs
+  |> Enum.each(fn {codepoint, code} ->
+    def encode_nucleotide(unquote(codepoint)), do: unquote(code)
+  end)
+
+  @pairs
+  |> Enum.each(fn {codepoint, code} ->
+    def decode_nucleotide(unquote(code)), do: unquote(codepoint)
```

Comment:
Do you expect the students to do meta-programming like this? I always try to write an example solution exactly like I expect the students to write it, to double-check that my assumptions about prerequisites are correct. But I'm not sure if that's a formal requirement of v3.

Reply:
Ya, I was just lazy to write out the clauses in case the pairs changed, which they did several times when writing the exercise. I'll write them out more completely.  In the stub, there is no trace of metaprogramming.